plugins/org.polarsys.capella.transition.system2subsystem/src/org/polarsys/capella/transition/system2subsystem/rules/fa/FunctionalChainInvolvementRule.java

```diff
@@ -60,79 +62,21 @@
 
   @Override
   public IStatus transformRequired(EObject element_p, IContext context_p) {
+    
     FunctionalChainInvolvement element = (FunctionalChainInvolvement) element_p;
-    // InvolvedElement must be transitioned
-
-    NamedElement involved = (NamedElement) element.getInvolved();
-
     IStatus result = TransformationHandlerHelper.getInstance(context_p).isOrWillBeTransformed(element.getInvolver(),
         context_p);
     if (!result.isOK()) {
       return result;
     }
-
-    result = TransformationHandlerHelper.getInstance(context_p).isOrWillBeTransformed(involved, context_p);
+    
     FunctionalChainAttachmentHelper helper = FunctionalChainAttachmentHelper.getInstance(context_p);
-
-    if (result.isOK()) {
-      if (!(helper.isValidElement(element, context_p) == Boolean.TRUE)) {
-        result = new Status(IStatus.WARNING, Messages.Activity_Transformation,
-            NLS.bind("Functional Chain Involvement ''{0}'' is not valid.", LogHelper.getInstance().getText(element_p)));
-      }
-
-    } else if (helper.isValidElement(element, context_p) == Boolean.TRUE) {
-      result = Status.OK_STATUS;
+    if (!(helper.isValidElement(element, context_p) == Boolean.TRUE)) {
+        return new Status(IStatus.WARNING, Messages.Activity_Transformation, NLS
+            .bind("Functional Chain Involvement ''{0}'' is not valid.", LogHelper.getInstance().getText(element_p)));
     }
 
-    if (result.isOK() && (involved instanceof FunctionalExchange)) {
```

Comment:
already looked while helper.isValidElement, and we didnt need to compute the real getNextValid, just that one of the element.getNextFunctionalChainInvolvements has to be valid.